src/components/confirmation-dialogs.js

```diff
@@ -104,3 +105,64 @@ export function confirmDocumentRevisionRestoration(t, revision, onOk, onCancel =
 
   dialog = confirm(createDialogProps());
 }
+
+export function confirmIdentityWithPassword({ t, username, userApiClient, onOk, onCancel = () => { } }) {
+  let dialog = null;
+  let createDialogProps = null;
+  let passwordValue = '';
+  let validationStatus = '';
+  let errorMessage = '';
+
+  const handleConfirmPassword = async password => {
+    try {
+      const { user } = await userApiClient.login({ username, password });
+      if (user) {
+        validationStatus = 'success';
+        errorMessage = '';
+        onOk();
+        return Promise.resolve();
+      }
+
+      validationStatus = 'error';
+      errorMessage = t('confirmationDialogs:wrongPasswordProvided');
+      dialog.update(createDialogProps());
+      return Promise.reject(new Error('Wrong password'));
```

Comment:
here just `throw new Error(...)` (but I'm not sure why we have to throw at all?)

Reply:
We need to reject the promise.

Reply:
True. Then we should throw without catching it (either outside the try or rethrow in the catch block). I'd rather not create promises manually inside async functions, it's a little bit confusing to see why.